packages/flame/lib/src/components/component.dart

```diff
@@ -27,14 +27,15 @@ abstract class Component {
   /// The smaller the priority, the sooner your component will be updated/rendered.
   /// It can be any integer (negative, zero, or positive).
   /// If two components share the same priority, they will probably be drawn in the order they were added.
-  final int priority;
+  int get priority => _priority;
+  int _priority;
 
   /// Whether this component should be removed or not.
   ///
   /// It will be checked once per component per tick, and if it is true, BaseGame will remove it.
   bool shouldRemove = false;
 
-  Component({this.priority = 0});
+  Component({int? priority = 0}) : _priority = priority ?? 0;
```

Comment:
I am almost sure that this will have the same effect and will prevent the user to pass null to the parameter

```suggestion
  Component({int priority = 0}) : _priority = priority;
```